integration/jnifuse/jnifuse-java/src/main/java/alluxio/jnifuse/LibFuse.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.jnifuse;
+
+import alluxio.jnifuse.utils.NativeLibraryLoader;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class LibFuse {
+
+  private enum LibraryState {
+    NOT_LOADED,
+    LOADING,
+    LOADED
+  }
+
+  private static AtomicReference<LibraryState> libraryLoaded
+      = new AtomicReference<>(LibraryState.NOT_LOADED);
+
+  static {
+    LibFuse.loadLibrary();
+  }
+
+  public native int fuse_main_real(AbstractFuseFileSystem fs, int argc, String[] argv);
+
+  public native ByteBuffer fuse_get_context();
+
+  public static void loadLibrary() {
+    if (libraryLoaded.get() == LibraryState.LOADED) {
+      return;
+    }
+
+    if (libraryLoaded.compareAndSet(LibraryState.NOT_LOADED,
+        LibraryState.LOADING)) {
+      String tmpDir = System.getenv("JNIFUSE_SHAREDLIB_DIR");
+      // loading possibly necessary libraries.
+      try {
+        System.loadLibrary("jnifuse");
+      } catch (UnsatisfiedLinkError e) {
+        // since it may be optional, we ignore its loading failure here.
+      }
+      try {
+        NativeLibraryLoader.getInstance().loadLibrary(tmpDir);
+      } catch (IOException e) {
```

Comment:
shall we handle UnsatisfiedLinkError or other possible Throwables here?

Reply:
NativeLibraryLoader#loadLibrary has catch and handle `UnsatisfiedLinkError`, the handle logic could throw IOException, so we catch IOException here and mark as NOT_LOADED. The following is `NativeLibraryLoader#loadLibrary `

```java
  public synchronized void loadLibrary(final String tmpDir) throws IOException {
    try {
      System.loadLibrary(sharedLibraryName);
    } catch (final UnsatisfiedLinkError ule1) {
      try {
        System.loadLibrary(jniLibraryName);
      } catch (final UnsatisfiedLinkError ule2) {
        loadLibraryFromJar(tmpDir);
      }
    }
  }
```